syntax = "proto3";

package protocols.communication;

import "protocols/common/feedback.proto";
import "protocols/common/game_command.proto";
import "protocols/common/message_type.proto";
import "protocols/common/robot_dribbler.proto";
import "protocols/common/robot_id.proto";
import "protocols/common/robot_kick.proto";
import "protocols/common/robot_velocity.proto";

message RobotInfo {
  repeated Output output = 1;
}

message Output {
  Command command = 1;
  common.Feedback feedback = 2;
}

message Command {
  common.MessageType msg_type = 1;
  common.RobotId robot_id = 2;

  common.GameCommand ref_command = 3;

  common.RobotVelocity robot_velocity = 4;

  common.RobotKick.KickCommand kick_command = 5;
  common.RobotDribbler.DribblerCommand dribbler_command = 6;

  Flags robot_flags = 7;
}

message Flags {
  bool robot_locked = 1;
  bool critical_move = 2;
  bool global_speed = 3;
}

message RobotCommand {
  repeated RobotCommand robot_command = 1;
}

message CommandOutput {
  uint32 id = 1;
  double vx = 2;
  double vy = 3;
  double vw = 4;
  bool front = 5;
  bool chip = 6;
  bool charge = 7;
  double kick_strength = 8;
  bool dribbler = 9;
  double dribbler_speed = 10;
}
